kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWindowedTable.java

```diff
@@ -198,6 +201,7 @@ public MongoWindowedTable(
     this.name = name;
     this.partitioner = partitioner;
     this.timestampFirstOrder = timestampFirstOrder;
+    this.retainDuplicates = partitioner.retainDuplicates();
```

Comment:
short answer: yes, it should be part of the table not the partitioner
long answer: right now it's a huge pain to get anything passed into the tables because of the way they get built from the table specs. It takes a LOT to add even one parameter this way -- see for example this PR where I just wanted the changelog topic name but piping it through was such a huge mess that [Almog suggested just tacking it onto the partitioner ](https://github.com/responsivedev/responsive-pub/pull/251#discussion_r1504573937). Obviously this situation is untenable since we can't throw everything into the partitioner, but it's nontrivial tech debt for another time. Hence I stuck with the partitioner approach for now